firestore/firestore_snippets/listen_test.go

```diff
@@ -36,66 +39,54 @@ func setup(ctx context.Context) (*firestore.Client, string) {
 
 func TestListen(t *testing.T) {
 	ctx := context.Background()
-	client, projectID := setup(ctx)
+	client, projectID := setup(ctx, t)
 	defer client.Close()
 
-	if projectID == "" {
-		t.Skip("Skipping firestore test. Set GOLANG_SAMPLES_FIRESTORE_PROJECT.")
+	cityCollection := []struct {
+		city, name, state string
+	}{
+		{city: "SF", name: "San Francisco", state: "CA"},
+		{city: "LA", name: "Los Angeles", state: "CA"},
+		{city: "DC", name: "Washington D.C."},
 	}
-	if err := prepareQuery(ctx, client); err != nil {
-		log.Fatalf("prepareQuery: %v", err)
+
+	for _, c := range cityCollection {
+		if _, err := client.Collection("cities").Doc(c.city).Set(ctx, map[string]string{
+			"name":  c.name,
+			"state": c.state,
+		}); err != nil {
+			log.Fatalf("Set: %v", err)
+		}
 	}
 	if err := listenDocument(ioutil.Discard, projectID); err != nil {
```

Comment:
Does this now take 30s per test? If so, could we make this more configurable, for example by allowing a ctx to be passed into the snippet also?

Reply:
+1. See other comment.